src/Surfnet/StepupMiddleware/ApiBundle/Doctrine/Type/UseRaLocationsOptionType.php

```diff
@@ -61,14 +61,20 @@ public function convertToPHPValue($value, AbstractPlatform $platform)
             return $value;
         }
 
+        // Prevent a too lenient boolean conversion on non-postgresql platforms
+        if ($platform->getName() !== 'postgresql') {
+            if ($value !== 1 && $value !== 0 && $value !== '1' && $value !== '0' && !is_bool($value)) {
```

Comment:
I have removed the strict checking and embraced conversion through DBAL.

As an aside, Doctrine [converts booleans to INT(1) for Oracle][1]. It's kind of funny as they [should only fallback][2] to something like that if booleans were not natively supported. But maybe it is because they want to "[ensure the least possible data storage requirements are met][2]" regardless of native support.

The reason I chose to exclude PostgreSQL specifically was that Doctrine's `$platform` only has a custom `convertFromBoolean` method if it is an instance of `PostgreSqlPlatform`, which IIRC deals with a list of TRUE/FALSE representations of strings. In all other scenarios, `$platform->convertFromBoolean($value)` would equal `return null === $item ? null: (bool) $item ;`. 

[1]: http://doctrine-orm.readthedocs.io/projects/doctrine-dbal/en/latest/reference/types.html#mapping-matrix
[2]: http://doctrine-orm.readthedocs.io/projects/doctrine-dbal/en/latest/reference/types.html#boolean